sources/common/cli.cpp

```diff
@@ -92,6 +92,19 @@ common::Cli::Cli()
             "Default value is false.\n"
             "--stale=<true|false>"
         )
+        (
+            "socks5",
+            value<bool>(),
+            "[OPTIONAL] Enable SOCKS5 proxy.\n"
+            "Default value is false.\n"
+            "--socks5=<true|false>."
+        )
+        (
+            "socks_port",
```

Comment:
Idea is good but multi parameters can be confuse for users.
Continue to use `--port` and keep `--socks5` remove `--socks_port`.

In the function `boost_endpoint    socksEndpoint{ {}, static_cast<boost::asio::ip::port_type>(config.mining.socksPort) };` replace `config.mining.socksPort` by `port` have the same effect.

The user must set `--port` and add `--socks5` to enable this feature.

Reply:
we can set `port` a `9050` by default if `--socks5` is enable and user did not define `--port`,

Reply:
I think we have some confusion here between `port` (the port of a stratum server) and `socks_port` (socks5 server running in localhost). Both of the parameters are need for connecting to a stratum via socks5 proxy server.

Nevertheless, I agree that the current parameter definition could be confusion for users. How about we use `--proxy=HOST:PORT` (following [xmrig](https://github.com/xmrig/xmrig)) and enable socks5 when `proxy` is not null?

Reply:
Ho ! My bad ok thanks ! I confused it. I'm ok with this solution. Please add a comment in this parameters for user.
Thanks you !

Reply:
Thanks for the review. I have pushed more detailed comments and examples regarding socks5 parameters.